fix: enzyme testing replaced by react tester

jciasenza opened the pull request:
Replaced enzyme display dropdown on toggling dropdown test in Main file for react-renderer tests.

I made PR of this issue https://github.com/openedx/edx-platform/issues/35245

And Testing OK

If something needs to be modified, let me know, thank you!!!
Atte
Juan Carlos (Aulasneo)

Developer Checklist

Test suites passing
Documentation and test plan updated, if applicable
Received code-owner approving review

Agrendalath committed:
fix: remove incorrect key type when filtering inaccessible blocks

Returning the list with a 'completion' string was added in 9bc0f85. However,
the `get_accessible_sequences` method expects a list of dicts with an 'id' key.

fix: enzyme testing replaced by react tester

andrey-canon committed:
feat: filter by current site organizations

Commit:
chore: Removed edx-token-utils dep and moved necessary logic to the repo

Commit:
chore: Run make compile-requirements to update dependencies

Commit:
chore: Removed unused dependency

irtazaakram committed:
fix: remove pkg_resources

Commit:
chore: fix tests

kdmccormick committed:
Revert "fix: Remove pointless Maintenance and Announcement apps (#35852)"

This reverts commit 9274852f2df9e64d3ee7b6de31227c3a5af4440b.

kdmccormick committed:
refactor: Remove broken Force-Publish UI from Studio

Co-Authored-By: Feanil Patel <feanil@axim.org>

ormsbee committed:
perf: reduce ORM queries needed for video quality info

The CourseQualityView used to call edx-val's get_video_for_course(),
which would return a fully serialized data structure that included all
encodings and inefficiently serialized them with many n+1 queries. This
is tolerable in a paginated web view, but not when pulling all of a
large courses's videos at once.

Making this change collapsed the number of queries for a large sample
MIT course from over 3000 down to 1.

Commit:
chore: Moved jwt file to openedx.core.lib

Commit:
chore: Updated TOKEN_SIGNING on cms

Commit:
chore: Updated defaults for token handling on CMS

kdmccormick committed:
revert: revert: refactor: Clean up lms/envs/production.py cruft

This reintroduces commit 15939232d547b751a06e9cb0c1c39ba78144bade,
which was reverted due to a typo.

The typo is fixed in the commit immediately following this one.

Co-Authored-By: Feanil Patel <feanil@axim.org>

feanil committed:
fix: Dropping terrain related config and docs.

The terrain app and its stub servers were removed as a part of
https://github.com/openedx/edx-platform/pull/36102 but these last few
things got missed.

kdmccormick committed:
fix: SHARED_kCOOKIE_DOMAIN -> SHARED_COOKIE_DOMAIN

ormsbee committed:
perf: update edx-when to 2.5.1 for performance fix

This addresses an issue where get_schedule_for_user was making an
excessive number of calls to the database.

katrinan029 committed:
feat: Upgrade Python dependency edx-enterprise

version bump

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

edx-requirements-bot committed:
chore: Upgrade Python requirements

ormsbee committed:
perf: upgrade XBlock to 5.1.1 for caching unknown tags

This should improve performance for courseware operations when there is
content that doesn't map to any existing XBlocks in the system. This
usually happens to old courses when custom XBlock types are deprecated
and removed, or when a course is imported from another instance with a
different set of installed XBlocks.

marlonkeating committed:
feat: Upgrade Python dependency edx-enterprise

fix: enterprise-customer-support should only return learners from given customer

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

nydelzaf committed:
fix: Replaced `mktemp` with a safer alternative `mkstemp`

* fix: Replaced `mktemp` with a safer alternative `mkstemp`

Signed-off-by: fazledyn-or <ataf@openrefactory.com>

---------

Signed-off-by: fazledyn-or <ataf@openrefactory.com>
Co-authored-by: Edward Zarecor <ezarecor@tcril.org>
Co-authored-by: Farhaan Bukhsh <farhaan@opencraft.com>

justinhynes committed:
feat: update default retirement states in devstack config (#36153)

We would like new devstack instances to better support the user retirement pipeline out of the box after provisioning. The first step is to update the default list of retirement states devstack is configured to add/use.

openedx-webhooks commented:
Thanks for the pull request, @jciasenza!


This repository is currently maintained by `@openedx/wg-maintenance-edx-platform`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>

<summary>:radio_button: Update the status of your PR</summary>

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.
</details>
<!-- comment:end_of_wip -->


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IHRydWV9 -->

jciasenza commented:
I don't understand, locally the changes pass all the tests and those that appear as failures are not files that have been modified.

robrap commented:
1. I'm re-running the failed tests. I think you just hit a flaky test failure.
2. You may still need to look into the quality failure.

jciasenza commented:
for example in this case, quality checks/Quality Others (ubuntu-24.04, 3.11, 20) (pull_request), I have not changed anything

Commit:
fix: Removed JWT constants from CMS and added comments on how to generate them

robrap commented:
You need to click the "Details" link to see what the actual failures are. There are ESlint failures.
```
Error: /home/runner/work/edx-platform/edx-platform/lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx: line 34, col 24, Error - Missing trailing comma. (comma-dangle)
Error: /home/runner/work/edx-platform/edx-platform/lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx: line 58, col 24, Error - Missing trailing comma. (comma-dangle)
Error: /home/runner/work/edx-platform/edx-platform/lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx: line 64, col 45, Error - Block must not be padded by blank lines. (padded-blocks)
Error: /home/runner/work/edx-platform/edx-platform/lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx: line 83, col 24, Error - Missing trailing comma. (comma-dangle)
```

Commit:
fix: ESlint failures

fix: remove enzyme dep and fix setupTest

fix: enzyme testing replaced by react tester

jciasenza commented:
@robrap Now they passed the tests!!! Thaks!!!

feanil committed:
build: Drop an unused Dockerfile (#36161)

* build: Drop an unused Dockerfile

We never cleaned up the docker file we were using for custom github
runners when we switched to standard runners.  This cleans that up and
removes another Docker mention that is out of date.

Co-authored-by: Kyle McCormick <kyle@axim.org>

robrap commented:
[inform] I create the following ticket to fix the flaky test failure issue:
- https://github.com/openedx/edx-platform/issues/36160

feanil committed:
docs: Update lms/envs/common.py

andrey-canon committed:
feat: add site-aware unit test

feanil commented on package.json:
```diff
@@ -105,6 +105,8 @@
+    "@testing-library/react": "^12.1.5",
+    "@testing-library/user-event": "^12.8.3",
     "babel-jest": "29.7.0",
     "enzyme": "3.11.0",
```
Shouldn't we be removing enzyme as a part of this change?

jciasenza replied:
If you're right, I didn't delete it thinking it was used in other components, but if it's not used I delete it.

feanil replied:
My understanding is that this is the only usage, and that was what was preventing us from migrating.  The end goal of the work should be to drop enzyme from edx-platform and make sure all the tests still pass.

jciasenza replied:
ok, now I delete it and do the push

jciasenza replied:
Do you think these dependencies should also be eliminated?

![image](https://github.com/user-attachments/assets/842057d6-0c9e-4f20-8952-40ea6457d4ac)

feanil replied:
Yes, I think all enzyme related dependencies should be eliminated and the package-lock.json re-built.

jciasenza replied:
I think it's a little more complex, I'm going to try to solve it.

![image](https://github.com/user-attachments/assets/48a9db55-a533-4211-aabe-4de1235506d1)

jciasenza replied:
in setupTest.js this is the enzyme configuration

jciasenza replied:
I think it fixed the problems

UsamaSadiq committed:
fix: remove publish-ci-docker-image workflow file (#36164)

deborahgu committed:
feat!: Removing the long-deprecated legacy course_modes chooser (#36156)

* feat!: Removing the long-deprecated legacy course_modes chooser

`course_modes/choose.html`  (and its corresponding
`_upgrade_button.html`)  were specifically only used for the edge case
where an enterprise user found themselves in the non-enterprise learner
dashboard, and attempted to  enroll in a course outside of the
enterprise flow. Once upon a time, in a 2U-only workflow, the commerce
system would apply specific discounts for users within the said case.
That's no longer true, and it has never been true outside of this one
company.

Removing this template  cleans up a legacy version of a legacy page that
was, realistically, exclusively seen by  employees of 2U,
and nobody else.

Removes:
* The corresponding testsfor behavior only seen in the legacy page.
* A waffle flag since all cases route as if the flag is set: `VALUE_PROP_TRACK_SELECTION_FLAG`: `course_modes.use_new_track_selection`
* Some variables set in  `CourseModeView` which were only ever rendered in the legacy template (`title_content`, `has_credit_upsell`) have been removed from the class.
* There is a high likelihood that the class is still  a target for re-factoring now that the legacy view is gone, but I'm  hesitant to touch something which is not covered by previously existing tests, because the logic around what template gets rendered when is complex.

FIXES: APER-3779
FIXES: #36090

edx-requirements-bot committed:
chore: Upgrade Python requirements (#36166)

pwnage101 committed:
feat: Upgrade Python dependency edx-enterprise

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

AhtishamShahid committed:
fix: bypass edx ace for sending goal reminder email using ses (#36148)

pomegranited committed:
feat: add and switch to downstream-only fields [FC-0076] (#36158)

Adds the concept of "downstream-only" fields to the XBlock upstream sync logic.

Downstream-only fields are customizable fields set only on the downstream XBlock -- we don't keep track of the upstream field value anywhere on the downstream XBlock. Changes made to these fields in the upstream block are ignored, and if the link to the upstream block is severed, the downstream changes are preserved (not reset back to defaults, like the upstream-tracked customizable fields are).

The fields chosen as "downstream-only" are those related to scoring and grading.

The `max_attempts` field was previously a customizable field that tracked the upstream value. However, because it is scoring-related, it has been converted to a "downstream-only" field.

This change impacts course authors' use of library content in their courses.

UsamaSadiq committed:
fix: remove pylint constraint (#36169)

deborahgu committed:
fix: catch a possible exception in beta course configuration (#36172)

* fix: catch a possible exception in beta course configuration

when the learner is a beta tester, and the beta test has been set up with a large duration, the courseware djangoapp can return an overflow error on a timedelta call. In those circumstances, this defaults to returning an unmodified date.

FIXES: APER-3848

kdmccormick committed:
feat: dump_settings management command (#36162)

This command dumps the current Django settings to JSON for
debugging/diagnostics. The output of this command is for *humans*... it
is NOT suitable for consumption by production systems.

In particular, we are introducing this command as part of a series of
refactorings to the Django settings files lms/envs/* and cms/envs/*.
We want to ensure that these refactorings do not introduce any
unexpected breaking changes, so the dump_settings command will both help
us manually verify our refactorings and help operators verify that our
refactorings behave expectedly when using their custom python/yaml
settings files.

Related to: https://github.com/openedx/edx-platform/pull/36131

deborahgu committed:
chore: linting  as a separate commit (#36179)

I'm letting autoformat hit this file to make it match our current
standards before I actually make any code changes.

deborahgu committed:
feat: increase frequency of scheduled notify credentials (#36180)

`notify_credentials` has 2 ways of running.
1. The manual,  one-off method  which uses `--args_from_database`  to specify what should be sent.
2. [The automated method](https://github.com/openedx/edx-platform/blob/7316111b35c8db0b93665e00aa4071685d772ab3/openedx/core/djangoapps/credentials/management/commands/notify_credentials.py#L157-L159), which runs on a regular schedule, to catch anything which fell through the cracks.

The automated method does a certain amount of time/date math in order to calculate the entry point of the window based on the current runtime.  This is, I assume, why it has some hardcoded logic; it's not at all simple to have a `cron`-run  management command running on a regular cadence that can do the same time logic.

```py
        if options['auto']:
            options['end_date'] = datetime.now().replace(minute=0, second=0, microsecond=0)
            options['start_date'] = options['end_date'] - timedelta(hours=4)
```

However, it is not ideal that the actual time window of 4 hours is hardcoded directly into `edx-platform`.

This fix

* creates an overridable `NOTIFY_CREDENTIALS_FREQUENCY` that defaults to 14400 seconds (4 hours).
* pulls that frequency into the quoted code

Using seconds allows maximum flexibility.

FIXES: APER-3383

sameeramin committed:
fix: Handle None identity providers in `_user_has_social_auth_record`

kdmccormick committed:
fix: Exactly preserve legacy settings dicts; rm KEYS_WITH_MERGED_VALUES

In the near term, we wish to precisely preserve the existing values of
all Django settings exposed by lms/envs/production.py in order to avoid
breaking legacy Django plugins without a proper announcement.

That includes preserving the behavior of these old, redundant dicts:
* ENV_TOKENS
* AUTH_TOKENS
* ENV_FEATURES
* ENV_CELERY_QUEUES
* ALTERNATE_QUEUE_ENVS

Particularly, it means we need to ensure that updates to Django
settings are reflected in these dicts. The most reliable way to do that
is to change the yaml-loading logic so that these values are aliased
to the corresponding values in the global namespace rather than
deep-copied.

Finally, we remove KEYS_WITH_MERGED_VALUES from the global namespace,
and inline the remaining list. We have modified the list (specifically,
we dropped the no-op MKTG_URL_OVERRIDES). Plugins should not be counting
on the value of the list, so we remove it.

jciasenza commented:
Hello @brian-smith-tcril, when you can check, I think this PR would be resolved now, Thanks !

kdmccormick committed:
feat: Make dump for fns, classes more stable and helpful (#36188)

The `dump_settings` command currently prints out the raw `repr(...)`s for
defined functions, e.g.:

    "WIKI_CAN_ASSIGN": "<function CAN_ASSIGN at 0x74ce5e9b2020>",

In addition to being uninformative, these `at 0x74ce...` hashes change every
run, so they appear in the diff as having "changed" every time. With this
commit, here's what `dump_settings` will print out for a function instead:

    "WIKI_CAN_ASSIGN": {
        "module": "lms.djangoapps.course_wiki.settings",
        "qualname": "CAN_ASSIGN"
    },

pwnage101 committed:
feat: Upgrade Python dependency edx-enterprise

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

UsamaSadiq committed:
fix: enable pylint warnings (#36195)

* fix: enable pylint warnings

navinkarkera committed:
fix: swagger docs ref_name conflicts (#36189)

* fix: swagger docs ref_name conflicts

* fix: swagger auto doc errors

* chore: bumps openedx-learning==0.18.2

---------

Co-authored-by: Jillian Vogel <jill@opencraft.com>

rpenido committed:
fix: advanced editor styling on library authoring [FC-0076] (#36146)

Fixes the styles for the advanced editors (poll, survey, LTI Provider, etc). Updated the code if `xblock_v2/xblock_iframe.html` to use `course-unit-mfe-iframe-bundle.scss`

sameeramin committed:
test: added test cases for `_user_has_social_auth_record`

eemaanamir committed:
chore: added language to ace message sent event (#36155)

* chore: added language to ace message sent event

* fix: fixed lint errors

* chore: updated ace version

brian-smith-tcril commented on lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx:
```diff
@@ -14,29 +12,36 @@ describe('ProblemBrowser Main component', () => {
 
-    test('render with selected block', () => {
-        const component = renderer.create(
+    test('display dropdown on toggling dropdown', async () => {
```
It seems this test isn't quite living up to the "display dropdown on toggling dropdown" name.

The only `expect` I see in here is checking that *before* the user clicks the `toggleButton` 'Some expected block name' doesn't appear on the screen.

This should be updated to verify the dropdown is actually displayed when it is toggled.

jciasenza replied:
Check this out

brian-smith-tcril commented on lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx:
```diff
@@ -47,50 +52,41 @@ describe('ProblemBrowser Main component', () => {
+        const toggleButton = screen.getByRole('button', { name: 'Select a section or problem' });
+        await userEvent.click(toggleButton);
+
+        await userEvent.click(toggleButton);
```
I think an extra `expect` in between these to verify the dropdown actually opens and closes (as opposed to just not ever opening) would probably be good.

jciasenza replied:
Thanks @brian-smith-tcril for giving me the feedback, I will review those points and upload the changes again.

brian-smith-tcril reviewed (commented):
Overall this looks great! Super happy to see things moving to RTL!

One thing I noticed is that the new tests that are replacing the snapshot tests aren't verifying the "Create a report of problem responses" button exists.

I also left a couple comments in the tests themselves. Nothing big, but definitely a couple spots where the tests could be more robust.

Thank you so much for doing this!

pwnage101 committed:
feat: Upgrade Python dependency edx-enterprise

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

UsamaSadiq committed:
fix: enable pylint warnings (#36196)

fix: enzyme testing replaced by react tester

fix: enzyme testing replaced by react tester

fix: enzyme testing replaced by react tester

fix: enzyme testing replaced by react tester

fix: enzyme testing replaced by react tester

jciasenza commented:
Hello, @brian-smith-tcril,
I made more changes to see what you think how they work now.

ormsbee committed:
feat!: remove unused email_marketing app

Quoting the README:

> If you are reading this and the 'Maple' Open edX release is out, you
> can safely delete this whole djangoapp. It only exists to hold old
> model migrations and any post-Maple installation will no longer have
> any model in the database for this app.
>
> But for some minor historical context, this djangoapp used to hold
> some integration with sailthru that we no longer needed.

We've released Sumac. This is overdue.

brian-smith-tcril commented on lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx:
```diff
@@ -67,30 +68,67 @@ describe('ProblemBrowser Main component', () => {
+        const toggleButton = screen.getByRole('button', { name: 'Select a section or problem' });
+        await userEvent.click(toggleButton);
+        const blockName = screen.queryByText('Some expected block name');
+        expect(blockName).toBeNull();
```
This seems a bit odd to me, it's expecting the query for `'Some expected block name'` to be null both before *and* after clicking the toggle button?

jciasenza replied:
I don't know if the way I modified it is right, but the tests passed.

jciasenza replied:
I don't know if the way I modified it is right, but the tests passed.

sarina replied:
FYI just because tests pass doesn't mean that they're _correct_. It doesn't make logical sense that you'd be testing that something is null both before and after a click. That seems like it's testing for the wrong thing.

jciasenza replied:
Yes, thanks [sarina](https://github.com/sarina), I am still working on the issues that Brian told me about, I hope to solve them and I can confirm, I have already made other PRs that were merged, but well, this one is causing me more conflicts.

brian-smith-tcril commented on lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx:
```diff
@@ -67,30 +68,71 @@ describe('ProblemBrowser Main component', () => {
+        await userEvent.click(toggleButton);
+        await waitFor(() => {
+            expect(screen.queryByText('Some expected block name')).toBeNull();
+        });
```
This looks great now!

jciasenza replied:
Thaks you !!!

brian-smith-tcril replied:
Actually sorry it doesn't. I missed the fact that you're now checking for `'Select a section or problem'` to not be null, it should be `'Some expected block name'` that is not null when toggling the dropdown

jciasenza replied:
I had my doubts about what you told me but it gives me errors like it doesn't find "Some expected block name", but hey it's progress

jciasenza replied:
in this case 
 `expect(screen.queryByText('Select a section or problem')).not.toBeNull();`

This is where I get the error arrives null or not found` 'Some expected block name'
`
```
ProblemBrowser Main component › hide dropdown on second toggle
    expect(received).not.toBeNull()
    Received: null
```
can you think of a solution?

brian-smith-tcril replied:
After looking at this more my main question is: where is `'Some expected block name'` coming from? I don't see anything in the tests putting that string anywhere.

The previous implementation of this test just looked for `BlockBrowserContainer`, it'd be helpful to know what is actually being rendered in that.

jciasenza replied:
I don't know if it is a good solution but replace the texts with `data-testid="block-browser-container"`

pwnage101 committed:
feat: Upgrade Python dependency edx-enterprise

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

schenedx committed:
chore: add django admin binding for course_action_state django app (#36207)

robrap committed:
docs: explain default session expiration (#36126)

pwnage101 committed:
feat: Upgrade Python dependency edx-enterprise

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

brian-smith-tcril commented:
I think part of what is making this PR hard for me to review is a lack of understanding of what is actually being tested. If you could take some screenshots of this running in a browser that would provide a lot of missing context for me.

jciasenza commented:
Ok, it's really the enzyme test that is being replaced by the reaction tester. 

![image](https://github.com/user-attachments/assets/20589239-0145-4c8b-867d-2b001b64af38)

I think the tests are replaced and pass well.

![image](https://github.com/user-attachments/assets/d1cc7f09-91cd-481c-a630-9558bd3f4446)

kdmccormick committed:
feat!: A Better API for Derived Settings (#36192)

The Python API for declaring derived settings was confusing to the uninitiated
reader, and also prone to spelling mistakes. This replaces the API with one
that is more readable and more concise, and updates the implementation of
`derive_settings` to properly derive settings declared using the new API.

BREAKING CHANGE: The `derived` and `derived_collection_entry` function are
replaced with the `Derived` class. We do not expect those functions to have
been used outside of edx-platform, but if they are, this commit will cause them
to loudly ImportError.

Note that there should be NO change in behavior to the `derive_settings`
function, which we DO know to be used by some external edx-platform plugins.

Part of: https://github.com/openedx/edx-platform/issues/36215

brian-smith-tcril commented:
I don't see the dropdown in that screenshot.

> it's really the enzyme test that is being replaced by the reaction tester.

While that is true, the old tests relied on snapshots, where the new tests are manually validating the behavior. The part of this I'm not understanding is *what* the tests should be looking for in the dropdown when it is toggled. It's not clear to me why an element with the test id `block-browser-container`is what we want to test for.

jciasenza commented:
I was reading a little and I think this is a good summary of the tests: 
![image](https://github.com/user-attachments/assets/6fbc62bc-ec2c-45e7-8af5-0b4e0521858d)

jesperhodge committed:
Revert "Replace pkg_resources with importlib.resources" (#36208)

brian-smith-tcril commented on lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx:
```diff
@@ -28,15 +27,16 @@ describe('ProblemBrowser Main component', () => {
-        const tree = component.toJSON();
-        expect(tree).toMatchSnapshot();
+        expect(screen.getByRole('button', { name: 'Select a section or problem' })).toBeInTheDocument();
     });
```
This test used to use a snapshot. That snapshot resulted in the following output
```
exports[`ProblemBrowser Main component render with basic parameters 1`] = `
<div
  className="problem-browser-container"
>
  <div
    className="problem-browser"
  >
    <button
      className="btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      Select a section or problem
    </button>
    <input
      disabled={true}
      hidden={false}
      name="problem-location"
      type="text"
      value={null}
    />
    <button
      className="btn"
      name="list-problem-responses-csv"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      Create a report of problem responses
    </button>
    <div
      aria-live="polite"
      className="report-generation-status"
    />
  </div>
</div>
`;
```

This test is now *only* checking for the `'Select a section or problem'` button. That is a large change in behavior from before.

### To merge this PR this test needs to be updated to test *everything* it used to be testing.

brian-smith-tcril commented on lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx:
```diff
@@ -47,16 +47,17 @@ describe('ProblemBrowser Main component', () => {
-        const tree = component.toJSON();
-        expect(tree).toMatchSnapshot();
+        expect(screen.getByRole('button', { name: 'Select a section or problem' })).toBeInTheDocument();
     });
```
This test *also* used to use a snapshot. That snapshot resulted in the following output

```
exports[`ProblemBrowser Main component render with selected block 1`] = `
<div
  className="problem-browser-container"
>
  <div
    className="problem-browser"
  >
    <button
      className="btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      Select a section or problem
    </button>
    <input
      disabled={true}
      hidden={false}
      name="problem-location"
      type="text"
      value="some-selected-block"
    />
    <button
      className="btn"
      name="list-problem-responses-csv"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      Create a report of problem responses
    </button>
    <div
      aria-live="polite"
      className="report-generation-status"
    />
  </div>
</div>
`;
```

This test is now *only* checking for the `'Select a section or problem'` button. That is a large change in behavior from before.

### To merge this PR this test needs to be updated to test *everything* it used to be testing.

brian-smith-tcril commented on lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx:
```diff
@@ -67,30 +68,70 @@ describe('ProblemBrowser Main component', () => {
+        const toggleButton = screen.getByRole('button', { name: 'Select a section or problem' });
+        await userEvent.click(toggleButton);
+        expect(fetchCourseBlocksMock).toHaveBeenCalledTimes(1);
     });
```
:tada:  This test does a perfect job of replicating the behavior from before using RTL! :tada:

kdmccormick committed:
refactor: Use `derive_settings` to lazy load settings. (#36205)

Some of our settings depend on the values of other settings.  Rather
than explicitly looking up each one in the YAML settings file, we can
simply derive them based on the setting in the YAML file after all the
YAML settings have been loaded.

Part of: https://github.com/openedx/edx-platform/issues/36215
Co-Authored-By: Feanil Patel <feanil@axim.org>

brian-smith-tcril commented on lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx:
```diff
@@ -67,30 +68,70 @@ describe('ProblemBrowser Main component', () => {
+        await waitFor(() => {
+            expect(screen.findByTestId('block-browser-container')).resolves.toBeInTheDocument();
+        });
+    });
```
This test is not properly replicating the behavior from before. The previous test checked for:
```js
        expect(component.find(BlockBrowserContainer).length).toBeFalsy();
        component.find(Button).find({ label: 'Select a section or problem' }).simulate('click');
        expect(component.find(BlockBrowserContainer).length).toBeTruthy();
```

Given the changes made in this PR to the `BlockBrowserContainer` component in `lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.jsx`, it makes sense to consider

```js
screen.findByTestId('block-browser-container')
```

to be serving the same purpose

```js
component.find(BlockBrowserContainer)
```

did before.

That being said, before it was testing

## In the Enzyme test 
```js
        expect(component.find(BlockBrowserContainer).length).toBeFalsy();
        component.find(Button).find({ label: 'Select a section or problem' }).simulate('click');
        expect(component.find(BlockBrowserContainer).length).toBeTruthy();
```
### Before Toggle
```js
expect(component.find(BlockBrowserContainer).length).toBeFalsy();
```
#### `BlockBrowserContainer` *IS NOT* in document.

### After Toggle
```js
expect(component.find(BlockBrowserContainer).length).toBeTruthy();
```
#### `BlockBrowserContainer` *IS* in document.

## In the new RTL test
### Before Toggle
```js
expect(screen.queryByTestId('block-browser-container')).toBeNull();
```
#### `BlockBrowserContainer` *IS NOT* in document.

### After (*FIRST*) Toggle
```js
        await waitFor(() => {
            expect(screen.findByTestId('block-browser-container')).resolves.toBeInTheDocument();
        });
```
#### `BlockBrowserContainer` *IS* in document.

### After (*SECOND?*) Toggle
```js
        await waitFor(() => {
            expect(screen.findByTestId('block-browser-container')).resolves.toBeInTheDocument();
        });
```
#### `BlockBrowserContainer` *IS (STILL?)* in document.

### The fact that we are finding an element with the test id `block-browser-container` *even after a second toggle* is very concerning to me. The test passing makes me think the test is not properly testing the behavior.

brian-smith-tcril commented on lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx:
```diff
@@ -67,30 +68,70 @@ describe('ProblemBrowser Main component', () => {
+        await waitFor(() => {
+            expect(screen.queryByText('block-browser-container')).toBeNull();
+        });
     });
```
This test did not exist before, and is a great addition!

That being said, this implementation is concerning, mostly because it is using `findByText` and `queryByText` when looking for a `data-testid`.

```js
        await waitFor(() => {
            expect(screen.findByText('block-browser-container')).resolves.toBeInTheDocument();
        });
```
### The fact that this is passing worries me. `findByText` should not be finding `block-browser-container` as that is a `data-testid`. `findByTestId` will find elements based on `data-testid`.

```js
        await waitFor(() => {
            expect(screen.queryByText('block-browser-container')).toBeNull();
        });
```

### It does not surprise me that this is passing, but that is because using `queryByText` to find a `data-testid` will *always* be null.

brian-smith-tcril reviewed (changes requested):
I have gone through and documented the issues I see with each of the tests in here. If you have questions about the issues I found please let me know.

DanielVZ96 committed:
feat: add publish status to library meilisearch index [FC-0076] (#36031)

Adds the publish status field to the libraries v2 meilisearch index in order to support filtering by component publish status: published, modified, never.

feanil committed:
build: Start watching the production.py files.

We want to make sure that changes to these align to the new refactors
we're doing so get tagged on changes to these files.

rayzhou-bit committed:
2u/course optimizer (#35887)

Commit:
fix: styling fix for course optimizer legacy tool link

AhtishamShahid committed:
feat: added events in account level notification preferences (#36231)

DanielVZ96 committed:
fix: static assets used in problem bank and library content block (#36173)

Static assets were not being copied into the course when using library content via Problem Bank or "Add Library Content" workflows.

fix enzyme testing

fix: enzyme testing

deborahgu committed:
feat!: Legacy account, profile, order history removal (#36219)

* feat!: Legacy account, profile, order history removal

This removes the legacy account and profile applications, and the order
history page. This is primarily a reapplication of #31893, which was
rolled back due to prior blockers.

FIXES: APER-3884
FIXES: openedx/public-engineering#71


Co-authored-by: Muhammad Abdullah Waheed <42172960+abdullahwaheed@users.noreply.github.com>
Co-authored-by: Bilal Qamar <59555732+BilalQamar95@users.noreply.github.com>

timmc-edx committed:
test: Add mock configs for testing production.py (#36237)

The values are nonsensical but this should be a fairly complete collection
of settings, both at the top level and in sub-dictionaries.

While edxapp will not work properly with these settings, the following
commands should work:

```
DJANGO_SETTINGS_MODULE=lms.envs.production LMS_CFG=lms/envs/mock.yml ./manage.py lms dump_settings
DJANGO_SETTINGS_MODULE=cms.envs.production CMS_CFG=cms/envs/mock.yml ./manage.py cms dump_settings
```

rayzhou-bit committed:
revert: remove problematic test code (#36238)

farhan committed:
chore: Adds fallback values to the xmodule xblocks global css variables (#36183)

* chore: Adds fallback values to the xmodule builtin blocks global css variables

jciasenza commented:
I returned the previous tests as they were and changed only the one that used Enzyme

Main Differences
Rendering Method
First test uses render() from @testing-library/react, which mounts the full component in an environment similar to a real browser.
Second test uses shallow() from Enzyme, which only renders the top-level component without its child components (shallow rendering).
Interactions with the DOM
First test interacts with the button using userEvent.click() and verifies DOM changes with screen.findByTestId().
Second test uses simulate('click') on a button inside the Main component and checks for the appearance/disappearance of BlockBrowserContainer in the React tree.
Handling Asynchronous Tests
First test uses await waitFor() to handle asynchronous DOM updates, which is crucial if the component has useEffect or state updates.
Second test is entirely synchronous and assumes that component changes happen immediately after simulate('click').
Checking Dropdown Visibility
First test uses screen.queryByTestId() and screen.findByTestId() to check if the element exists in the DOM.
Second test uses component.find(BlockBrowserContainer).length to count whether the component was rendered or not.

deborahgu committed:
APER-3884: make sure links point to MFE not legacy (#36241)

now that the legacy profile and account pages have been removed, we need to make sure that all of the links point to the MFE URLs; we were relying on the legacy applications to do redirection before.

FIXES: APER-3884
FIXES: openedx/public-engineering#71

navinkarkera committed:
feat: upstream downstream link model [FC-0076] (#36111)

Adds models, tasks, signal handlers and api's for adding, updating and deleting upstream->downstream links in database.

sarina commented:
@angonz could we get your eyes (or another Aulasneo dev) on this PR? Brian's outstanding comments aren't addressed - we need these tests refactored in such a way that they need to test what they were testing previously. Brian has explained why this is the case and what is wrong. 

@jciasenza could you please put this PR in draft until your team is ready for review of tests that match what the snapshots were previously testing? Thank you!

PKulkoRaccoonGang committed:
chore: iframe rendering optimization (#35959)

Iframe reload optimizations for various xblock related actions. Added some improvements related to scrolling to the current xblock. Fixed behavior of the xblock action dropdown list.

AhtishamShahid committed:
fix: resolved issues with ses impl in goal reminder email (#36233)

Commit:
fix: studio edx.org-next theme

Attaching jira ticket for reference: https://2u-internal.atlassian.net/browse/TNL-11731

- compile_sass script was generating the `studio-main-v1.scss` file but it wasn't correct and wasn't picking the changes from paragon. As a result, `edx.org-next` theme wasn't being applied to the studio. And the reason was this line: https://github.com/openedx/edx-platform/pull/35233/files#:~:text=//%20%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D-,%40import%20%27_builtin%2Dblock%2Dvariables%27%3B,-%40import%20%27bourbon.
- This PR fixes the studio theme as I have rearranged `studio-main-v1.scss` file so that theme changes can be applied correctly.

rpenido committed:
fix: error when loading survey block without user in new runtime (#36226)

* fix: always define a student_data_store to prevent errors on XBlock load

* chore: bump xblock version to 5.1.2

sarina commented:
@jciasenza see the following guides about PR tests, it'll explain why "Lint Commit Messages" is failing and gives more detail about reading build output.

* https://docs.openedx.org/en/latest/developers/references/running_pr_tests.html
* https://docs.openedx.org/en/latest/developers/quickstarts/first_openedx_pr.html

katrinan029 committed:
feat: Upgrade Python dependency edx-enterprise

version bump

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

irtazaakram committed:
fix: replace pkg_resources with importlib.resources (#36213)

AhtishamShahid committed:
chore: added debug logs in instructor enrolment email (#36250)

ChrisChV committed:
fix: Advanced editor styles [FC-0076] (#36221)

* Adds padding to the text of the competition block editor.
* Fix the style of the Save and Cancel buttons of the PDF block editor.
* Moves the Save and Cancel buttons to the bottom of the modal.
* Moves the Save and Cancel button to the right.
* Which edX user roles will this change impact? "Course Author"

pomegranited committed:
Make copied tags editable again after breaking the upstream link to library content [FC-0076] (#36228)

When deleting an upstream library block, ensure that any tags that may have been copied to downstream blocks are made editable again. This is achieved by un-setting the `is_copied` flag on the downstream tags.

ChrisChV committed:
chore: Update requirements to use new version of xblock-google-drive (#36227)

AhtishamShahid committed:
fix: log only for allowed_enroll email (#36254)

michaelroytman committed:
feat: make computation of due date for edx-proctoring powered exams consistent

This commit fixes an inconsistency in the way an exam due date is computed for courses that do not use an LTI based proctoring provider.

The edx-exams microservice was released last year to provide support for LTI based proctoring providers. After the release of this microservice, all proctoring requests initiated by the platform began to be funneled through the microservice, which acted as a broker for these requests, routing them directly to the service for exams in courses using an LTI based proctoring provider or to the platform edx-proctoring plugin for all other cases.

There is an asynchronous task in the platform that syncs exams from the platform to either the edx-exams microservice or the edx-proctoring plugin. Prior to the release of the microservice, this task computed the due date on exams as the exam subsection due date if the course was instructor-paced or None. After the release of the microservice, the task computed due dates differently than before. The due date on exams was computed as the due date on the exam, if there was one, or the end date of the course, if there was one, or None. This differed from the prior definition.

This resulted in inconsistent due date behavior. The exams in courses that were published or republished after the edx-exams microservice was released had the new computation of due date, while exams in courses that were published or republished before the edx-exams microservice was released had the old computation of due date.

This causes an issue for all exams in courses using non-LTI based providers. This is because the due date on exams across all courses that either do not use proctoring or that use a non-LTI based provider are inconsistent, depending on when they were last published.

This commit reintroduces the old computation to the task for exams in courses using a non-LTI based proctoring provider (i.e. those courses whose exams are not powered by the edx-exams microservice). In order to maintain the functionality of edx-exams, we continue to compute the due date as before for exams in courses using an LTI based proctoring provider (i.e. those courses whose exams are powered by the edx-exams microservice).

ChrisChV committed:
chore: Bump version of xblock-poll to 1.14.1 (#36252)

Updates the requirements for using the latest version of xblock-poll. Changes are generated by make compile-requirements

feanil committed:
build: Add more apps to the set of installed apps.

These app models get pulled in when trying to build the OpenAPI docs so
we need their relevant apps installed into the environment used to
generate the docs.

feanil committed:
build: We don't need to override this setting.

The format of the `source_suffix` setting has change but the default is
fine for us so we don't need to override this.

feanil committed:
docs: Update the Open API docs.

syedsajjadkazmii committed:
feat: override refund_entitlement (#36248)

UsamaSadiq committed:
chore: update edx-proctoring package version (#36256)

* fix: fix tests with new version of edx-proctoring

* feat: Upgrade Python dependency edx-proctoring (#36255)

chore: update edx-proctoring version

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: UsamaSadiq <41958659+UsamaSadiq@users.noreply.github.com>

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

awais786 committed:
feat!: upgrading api to DRF. (#35609)

eemaanamir committed:
fix: fixed escaped characters in goal reminder email (#36260)

github-actions[bot] committed:
feat: Upgrade Python dependency faker (#36262)

chore: upgrade faker package version

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: UsamaSadiq <41958659+UsamaSadiq@users.noreply.github.com>
Co-authored-by: Usama Sadiq <usama.sadiq@arbisoft.com>

hassan-raza-1 committed:
fix: tracker context issue (#36247)

Co-authored-by: Hassan Raza <h.raza@192.168.1.12>

github-actions[bot] committed:
feat: Upgrade Python dependency geoip2 (#36266)

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

Co-authored-by: UsamaSadiq <41958659+UsamaSadiq@users.noreply.github.com>
Co-authored-by: Usama Sadiq <usama.sadiq@arbisoft.com>

navinkarkera committed:
feat: entity link view and api (#36190)

Adds api for listing upstream entity links for a given course.

farhan committed:
feat: Upgrade Python dependency xblocks-contrib

chore: Update xblocks-contrib pypi package

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

awais786 committed:
feat!: upgrading get_proctored_exam_results api to DRF ( 32 ) (#35612)

* feat!: upgrading api to DRF

AhtishamShahid committed:
fix: log only for allowed_enroll email (#36273)

ChrisChV committed:
fix: Render Word cloud block editor in libraries [FC-0076] (#36199)

* fix: Render Word cloud and conditional block editor

- The xmodule-type to render is MetadataOnlyEditingDescriptor
- The xmodule type `MetadataOnlyEditingDescriptor` renders a `<div>` with the block metadata in the `data-metadata` attribute. But is necessary to call `XBlockEditorView.xblockReady()` to run the scripts to build the editor using the metadata.
- To call XBlockEditorView.xblockReady() we need a specific require.config

* fix: Adding save and cancel button

* fix: save with studio_submit of conditional_block and word_cloud_block

* test: Tests for studio_submit of conditional and word cloud

* revert: Delete studio_submit of conditional block. It is not supported

* style: Fix lint

---------

Co-authored-by: Navin Karkera <navin@opencraft.com>

Commit:
fix: send correct status for scanning

navinkarkera committed:
feat: Upgrade Python dependency xblock-drag-and-drop-v2

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

PKulkoRaccoonGang committed:
feat: course unit - edit iframe modal window (#35777)

Adds logic to support the functionality of editing xblocks via the legacy modal editing window.

Commit:
Merge upstream into my branch

awais786 committed:
docs: fixing make docs command. (#36280)

* docs: fixing make docs command.

mariajgrimaldi committed:
fix: import authoring filter from content_authoring instead (#36109)

Correctly name authoring subdomain according to DDD docs: https://openedx.atlassian.net/wiki/spaces/AC/pages/663224968/edX+DDD+Bounded+Contexts

ChrisChV committed:
feat: Video editor supports transcripts [FC-0076] (#36058)

* Add error handler on save video to avoid creating sjson
* Support transcripts without edx_video_id in definition_to_xml
* When copying a video from a library to a course: Create a new edx_video_id
* Save transcripts as static assets in a video in a library when adding a new transcript.
* Delete transcripts as static assets in a video in a library when deleting transcripts.
* Support download transcript in a video in a library.
* Support replace transcript in a video in a library.
* Support updating transcripts in video in a library.
* Refactor the code of downloading YouTube transcripts to enable this feature in libraries.
* Support copy from a library to a course and a course to a library.

deborahgu committed:
feat: management command to purge information about web certificates (#36287)

for Open edX  operators who still have users with legacy PDF certificates, retirement requires first extracting information from the user's GeneratedCertificate record in order to delete the 4 associated files for each PDF certificate, and then removing the links to the relevant files.  this creates a management command to do that work.

After thinking about it, I have removed the update to `status` from this management command, as per the original specification of the ticket. I added it for completeness originally, but was already uncomfortable, because it's not exactly accurate. The `CertificateStatuses` enum does define a `deleted` status:

```
    deleted             - The PDF certificate has been deleted.
```

but I think it's inappropriate to use here.

#### Why not use `CertificateStatuses.deleted` in the first place

There are multiple places in the code where it's clear that almost all of the statuses are legacy and unused (eg. [Example 1](https://github.com/openedx/edx-platform/blob/6c6fd84e533e6cb272da2e618f227225ad907232/lms/djangoapps/certificates/data.py#L12-L34), [Example 2](https://github.com/openedx/edx-platform/blob/1029de5537752ecf0477d578dba8a932e741e497/common/djangoapps/student/helpers.py#L491-L492)). There are innumerable APIs in the system that have expectations about what might possibly be returned from a `GeneratedCertificate.status` object, and none of them is expecting `deleted`

#### Why not revoke the certificate

Ultimately, the certificate isn't revoked, which has a specific meaning around saying it was unearned. The certificate was earned; it has simply been deleted.  We should not be kicking off program certificate invalidation, because that's not what's happening. We should be trusting the normal user retirement process to remove/purge PII from any program certificates that might exist. The nature of web certificates simply means that we are going through this process outside of the normal retirement flow. The normal retirement flow can be trusted to implement any certificate object revocation/removal/PII-purging, and doing an extra step outside of that flow is counterproductive.

#### Why not robustly add a flow for `CertificateStatuses.deleted`

When PDF certificates were removed from the system, they weren't removed in their entirety. Instead, we have this vestigial remains of PDF certificates code, just enough to allow learners to display and use the ones that they already have, without any of the support systems for modifying them. Adding a `deleted` status, verifying that all other APIs wouldn't break in the presence of a certificate with that status, adding the signals to process and propagate the change: all of this would be adding more tech debt upon the already existing technical debt which is the PDF certs code. Better to simply add this one necessary data integrity change, and focus on a process which might allow us to eventually remove the web certificates code.

#### Why it is good enough to ignore the status

The original ask was simply to enforce data integrity: to remove links to files that have been deleted, as an indication that they've been deleted. I only added `status` update out of a (misplaced but well-intentioned) completionist urge.

FIXES: APER-3889

BryanttV committed:
refactor!: remove `ENABLE_HOME_PAGE_COURSE_API_V2` feature toggle (#36181)

fix: reformulation of the Main component test

fix: reformulation of the Main component test

fix: reformulation of the Main component test

fix: reformulation of the Main component test

fix: reformulation of the Main component test

fix: reformulation of the Main component test

fix: reformulation of the Main component test

fix: enzyme testing replaced by react tester

fix: enzyme testing replaced by react tester

fix: enzyme testing replaced by react tester

AfaqShuaib09 committed:
chore: add TRANSCRIPT_LANG_CACHE_TIMEOUT setting variable in common setting envs (#36296)

* chore: add TRANSCRIPT_LANG_CACHE_TIMEOUT setting variable in common setting envs
* chore: bump edx-val version

fix: enzyme testing replaced by react tester

fix: enzyme testing replaced by react tester

fix: enzyme testing replaced by react tester

fix: enzyme testing replaced by react tester

fix: enzyme testing replaced by react tester

jciasenza committed:
Merge branch 'openedx:master' into jci/issue35245

jciasenza committed:
Merge branch 'openedx:master' into jci/issue35245

jciasenza commented:
Hello @feanil, @brian-smith-tcril and @sarina :  I have decided to close this PR, and after carefully studying the problems I have had with it, I have decided to start a new PR where I explain everything I have done. I hope that everything requested now works.
According to what I discussed with my team, it seemed like the most logical thing to do after the errors I had after the updates. 
A thousand apologies and thanks for reviewing it. #36303
